yb-voyager/src/compareperf/comparison_test.go

```diff
@@ -0,0 +1,269 @@
+	for _, comp := range topBySlowdown {
+		assert.Equal(t, MATCHED, comp.MatchStatus, "All results should be MATCHED")
+		assert.InDelta(t, expectedSlowdownRatio, comp.SlowdownRatio, 0.001)
+	}
+}
+
+// TestGetTopBySlowdown_TargetFaster tests when target is faster (slowdown < 1)
+func TestGetTopBySlowdown_TargetFaster(t *testing.T) {
+	// Create comparator where target is 20% faster, 0.8 = 20% faster
+	comparator := createMockComparator(2, 0, 0, 0.8)
+	expectedSlowdownRatio := (8.0 + SLOWDOWN_RATIO_OFFSET) / (10.0 + SLOWDOWN_RATIO_OFFSET)
+	allComparisons := comparator.matchQueries()
+	for _, comp := range allComparisons {
+		comp.calculateMetrics()
+	}
+
+	topN := 10
+	topBySlowdown := comparator.getTopBySlowdown(allComparisons, topN)
+
+	// assertions
+	assert.Len(t, topBySlowdown, 2, "Should return both matched queries")
+	for _, comp := range topBySlowdown {
+		assert.Equal(t, MATCHED, comp.MatchStatus, "Should be MATCHED")
+		assert.InDelta(t, expectedSlowdownRatio, comp.SlowdownRatio, 0.001)
+		assert.True(t, comp.SlowdownRatio < 1.0, "Slowdown should be < 1 when target is faster")
+	}
+}
+
+// TestGetTopByImpact tests the getTopByImpact method with predictable mock data
+func TestGetTopByImpact(t *testing.T) {
+	// Create comparator with 30 matched queries, target 50% slower
+	comparator := createMockComparator(30, 10, 10, 1.5)
+	expectedImpactScore := 15.0*QUERY_CALLS_FOR_TEST - 10.0*QUERY_CALLS_FOR_TEST
+	allComparisons := comparator.matchQueries()
+	for _, comp := range allComparisons {
+		comp.calculateMetrics()
+	}
+
+	topN := 10
+	topByImpact := comparator.getTopByImpact(allComparisons, topN)
+
+	// assertions
+	assert.Len(t, topByImpact, topN, "Should return only the %d matched queries", topN)
+	for i, comp := range topByImpact {
+		assert.Equal(t, MATCHED, comp.MatchStatus, "Query %d should be MATCHED", i)
+		assert.Equal(t, expectedImpactScore, comp.ImpactScore)
+		assert.Contains(t, comp.Query, "Query-", "Should be a matched query")
+	}
+	for i := 1; i < len(topByImpact); i++ {
+		assert.True(t, topByImpact[i-1].ImpactScore >= topByImpact[i].ImpactScore,
+			"Results should be sorted by impact score in descending order")
+	}
+}
+
+// TestGetTopByImpact_WithTopNLimit tests the limit parameter behavior
+func TestGetTopByImpact_WithTopNLimit(t *testing.T) {
+	// Create comparator with 50 matched queries, target 2x slower
+	comparator := createMockComparator(50, 0, 0, 2.0)
+	expectedImpactScore := 20.0*QUERY_CALLS_FOR_TEST - 10.0*QUERY_CALLS_FOR_TEST
+	allComparisons := comparator.matchQueries()
+	for _, comp := range allComparisons {
+		comp.calculateMetrics()
+	}
+
+	topN := 3
+	topByImpact := comparator.getTopByImpact(allComparisons, topN)
+
+	// assertions
+	assert.Len(t, topByImpact, topN, "Should return only the %d matched queries", topN)
+	for _, comp := range topByImpact {
+		assert.Equal(t, MATCHED, comp.MatchStatus, "All results should be MATCHED")
+		assert.Equal(t, expectedImpactScore, comp.ImpactScore)
+	}
+}
```

Comment:
Same as above. This might be redundant if we are already providing and checking limits in every test.